test/conditional/test_licenses.sh

```diff
@@ -35,4 +35,13 @@ do
     fi
 done
 
+yarnlicense = `yarn licenses list | grep -iP '(?<!OR )(?<!L)GPL(?! OR)'`
+
+if yarnlicense && true
```

Comment:
I think this is slightly different than the source where you copied it from, since the above `&& true` is apart of the `if` statement which won't trigger the script halt on a non-zero exit code with `set -e`. So if it were to fail, it wouldn't go any farther than line 38 because that isn't in the `if` test and `set -e`. You should get the following error if it actual goes past line 38 as it is:
```
./test/conditional/test_licenses.sh: line 41: yarnlicense: command not found
```

Reply:
Yeah, added spaces between `yarnlicenses=`... which made it interpret it not as a variable assignment, should be updated now.

Reply:
@rtibbles Note that the error I shared is line 41, not line 38. When `yarn licenses list | grep -iP '(?<!OR )(?<!L)GPL(?! OR)'` is executed, if it doesn't find a match it will exit with code `1` which stops the script at line 38 and exits with the same code:
```
$ ./test/conditional/test_licenses.sh
Checking all requirements installed with pip, except docutils;nose;...
$ echo $?
1
```
If I substitute `GPL` in the regex for `MIT`, then it should match, which allows it to get to line 41, and thus:
```
$ ./test/conditional/test_licenses.sh
Checking all requirements installed with pip, except docutils;nose;...
./test/conditional/test_licenses.sh: line 41: yarnlicense: command not found
```

Reply:
Ah, gotcha - yeah, that's what is causing the exit code 1 then!

Reply:
I finally trial and errored my way to success.

Ran the script locally using MIT instead of GPL to confirm it does indeed give a non-zero exit code in that case.